### ISSUE TITLE
Unable to render chat

### ISSUE THREAD

**Issue by Keagel** [Zibbp/ganymede#61]
Hello,

I come to you with a second issue. Up until now I had never enabled chat archiving. Now that I have, I'm noticing that I always get the same error when rendering the chat:

```
Failure processing application bundle.
Failed to determine location for extracting embedded files.
DOTNET_BUNDLE_EXTRACT_BASE_DIR is not set, and a read-write cache directory couldn't be created.
```

I looked into the code and I assume the error comes from TwitchDownloaderCLI, but it seems more likely that it is an issue with my setup. I'm overriding the user of the container to make it fit the UID/GID of my `mergerfs` mount as otherwise `ganymede` can't write to the folder so I wouldn't be surprised if it was a permission issue (especially with the "read-write cache directory" message), though the `/tmp` folder I have mounted is writable by any user. Considering the same folder is used to download videos and that part working fine, I'm not sure how to debug further.

Any ideas?

**Comment by Zibbp** [comment]
Hi,
Definitely appears to be a permission issue. To confirm the mounted `tmp` folder was given access to everyone with something like `chmod -R 777 /host/path/to/mounted/tmp` ? Are you able to exec into the `ganymede-api` using your custom user and create files in the `/tmp` folder?

It appears that the TwitchDownloaderCLI can take a `--temp-path` argument. Try adding this to the "chat render args" in Admin > Settings setting it to a path that is **not mounted** and local to the container such as. `--temp-path=/tdl/tmp`. Though it may not work because the user probably doesn't have permissions to create a root folder like that. Or maybe even try setting it to the `tmp` folder, maybe it got confused with your custom user? `--temp-path=/tmp`.

**Comment by Keagel** [comment]
> To confirm the mounted tmp folder was given access to everyone with something like chmod -R 777 /host/path/to/mounted/tmp ? Are you able to exec into the ganymede-api using your custom user and create files in the /tmp folder?

That's right. Folder is set to chmod 777 for testing purposes. The `/tmp` folder can be written into when I exec into the container, but for some reason TwitchDownloader can't.

> It appears that the TwitchDownloaderCLI can take a --temp-path argument. Try adding this to the "chat render args" in Admin > Settings setting it to a path that is not mounted and local to the container such as. --temp-path=/tdl/tmp. Though it may not work because the user probably doesn't have permissions to create a root folder like that. Or maybe even try setting it to the tmp folder, maybe it got confused with your custom user? --temp-path=/tmp.

I've tried that and unfortunately since all folders inside the container are created by the `root` user, none of them can be written into by the regular user. Commenting out my custom user from the container fixes the problem, confirming it's a permission issue.

**Comment by Zibbp** [comment]
I did some testing and I was able to reproduce the error by setting a custom `user: 1000:1000`. Did some research regarding the error and came across a post about manually setting the `DOTNET_BUNDLE_EXTRACT_BASE_DIR` env var. I did so by adding
```
- DOTNET_BUNDLE_EXTRACT_BASE_DIR=/tmp
```
To the `docker-compose.yml` file for the API container. Upon testing again, it completed without issue. I know nothing about c# or the .net framework so I can't tell you why this fixed it, but it did for me.

Give that a try and let me know if it works.
